shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java

```diff
@@ -0,0 +1,234 @@
+      + "[filter arg] can be one of the following:\n"
+      + "    -live                         Live workers\n"
+      + "    -lost                         Lost workers\n"
+      + "    -worker <worker_ip_addresses> Specified workers, IP addresses separated by \",\"\n";
+
+  private BlockMasterClient mBlockMasterClient;
+  private PrintStream mPrintStream;
+  private int mIndentationLevel = 0;
+  private StringBuilder mStringBuilder;
+
+   /**
+   * Creates a new instance of {@link CapacityCommand}.
+   *
+   * @param blockMasterClient client to connect to block master
+   * @param printStream stream to print summary information to
+   */
+  public CapacityCommand(BlockMasterClient blockMasterClient, PrintStream printStream) {
+    mBlockMasterClient = blockMasterClient;
+    mPrintStream = printStream;
+    mStringBuilder = new StringBuilder();
+  }
+
+  /**
+   * Runs report capacity command.
+   *
+   * @param cl CommandLine to get client options
+   * @return 0 on success, 1 otherwise
+   */
+  public int run(CommandLine cl) throws IOException {
+    if (cl.hasOption("h")) {
+      System.out.println(USAGE);
+      return 0;
+    }
+    ReportWorkerOptions options = getOptions(cl);
+    printWorkerCapacityInfo(options);
+    return 0;
+  }
+
+  /**
+   * Prints Alluxio capacity information.
+   *
+   * @param options ReportWorkerOptions to define the report worker info range
+   */
+  public void printWorkerCapacityInfo(ReportWorkerOptions options) throws IOException {
+    List<ReportWorkerInfo> workerInfolist = mBlockMasterClient.getReportWorkerInfoList(options);
+
+    // Summarize the worker capacity information
+    long sumCapacityBytes = 0;
+    long sumUsedBytes = 0;
+    Map<String, Long> sumTotalBytesOnTiersMap = initializeTierMap();
+    Map<String, Long> sumUsedBytesOnTiersMap = initializeTierMap();
+
+    // Cache the worker capacity information
+
+    for (ReportWorkerInfo workerInfo : workerInfolist) {
+      mStringBuilder.append(String.format("%-20s %-20s %-20s %-20s %-20s%n",
+          "Node Name", "Last Heartbeat", "Workers Capacity", "Space Used", "Used Percentage"));
+
+      long usedBytes = workerInfo.getUsedBytes();
+      long capacityBytes = workerInfo.getCapacityBytes();
+      String usedPercentageInfo = "Unavailable";
+      if (capacityBytes != 0) {
+        int usedPercentage = (int) (100L * usedBytes / capacityBytes);
+        usedPercentageInfo = String.format("%s%%", usedPercentage);
+      }
+      mStringBuilder.append(String.format("%-20s %-20s %-20s %-20s %-20s%n",
+          workerInfo.getAddress().getHost(),
+          workerInfo.getLastContactSec(),
+          FormatUtils.getSizeFromBytes(capacityBytes),
+          FormatUtils.getSizeFromBytes(usedBytes),
+          usedPercentageInfo));
+
+      sumCapacityBytes += capacityBytes;
+      sumUsedBytes += usedBytes;
+
+      Map<String, Long> totalBytesOnTiers = workerInfo.getCapacityBytesOnTiers();
```

Comment:
this method is pretty long, perhaps we could do the aggregation in a helper method instead

Reply:
Change structure to:
run() {call generateCapacityReport()}
generateCapacityReport() { // For easy testing
    collectWorkerInfo(options);
    printAggregatedInfo(options);
}